native/native_main.nim

```diff
@@ -0,0 +1,240 @@
+# compile with 
+# nim c --verbosity:0 native_main.nim
+
+# test with
+# time printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | ./native_main
+
+# compare with
+# time printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | python ~/.local/share/tridactyl/native_main.py
+
+import json
+import options
+import osproc
+import streams
+import os
+import posix
+# import endians
+import struct # nimble install struct
+
+const VERSION = "0.2.0"
+
+type 
+    MessageRecv* = object
+        cmd*, version*, content*, error*, command*, variable*, file*, dir*, to*, origin*: Option[string]
+        code: Option[int]
+type 
+    MessageResp* = object
+        cmd*, version*, content*, error*, command*: Option[string]
+        code: Option[int]
+
+# let a = MessageResp(cmd: some(""))
+# echo(a)
+
+proc trySwapJsonKey(json: JsonNode, old: string, nouveau: string) =
+    try:
+        json[nouveau] = json[old]
+        delete(json, old)
+    except KeyError:
+        discard
+
+
+proc getMessage(strm: Stream): MessageRecv =
+
+    try:
+        var length: int32
+        read(strm,length)
+        write(stderr, "Reading message length: " & $length & "\n")
+        if length == 0:
+            write(stderr, "No further messages, quitting.\n")
+            close(strm)
+            quit(0)
+
+        let message = readStr(strm, length)
+        write(stderr, "Got message: " & message & "\n")
+        var raw_json = parseJson(message)
+
+        # Compatibility with Python native messenger:
+        # rename env's _var_ key to _variable_ coz _var_ is reserved in Nim
+        trySwapJsonKey(raw_json, "var", "variable")
```

Comment:
This isn't necessary, because Nim has [stropping](https://nim-lang.org/docs/manual.html#lexical-analysis-stropping), e.g.:
```nim
msg.`var`.get()
```

Reply:
Oh, nice, thanks. I've never heard of that before. I thought Nim was being a bit greedy reserving common variable names : )

Reply:
I only found out about it pretty much by pure chance, but I didn't expect it to actually come in handy.